src/lib/response/responseUtil.ts

```diff
@@ -0,0 +1,52 @@
+// Eslint doesn't like TypeScript enums
+/* eslint-disable no-unused-vars */
+
+import { NextResponse } from 'next/server';
+
+export type MessageResponseType = {
+  message: string;
+  messageType: MessageType;
+  statusCode?: StatusCodeType;
+};
+
+type MessageWithDataResponseType = MessageResponseType & {
+  data: any;
+};
+
+export enum MessageType {
+  ERROR = 'error',
+  WARNING = 'warning',
+  INFO = 'info',
+  SUCCESS = 'success',
+}
+
+// https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/
+export enum StatusCodeType {
+  OK = 200,
+  CREATED = 201,
+  BAD_REQUEST = 400,
```

Comment:
I've actually also been using '422 Unprocessable Content' when returning an error message about the user trying to create a duplicate tag. (The request isn't 'Bad' per se, it's just that a duplicate already exists in the database).

Reply:
Yeah, it can simply be added to that list